### PR TITLE
Add disability and wheelchair questions validation

### PR DESCRIPTION
**Pull request by ghost**
Add disability and wheelchair questions validation
Add an helper to easily check if a page is an interruption screen

https://digital.dclg.gov.uk/jira/browse/CLDC-1586

### CODE REVIEW

File: app/models/form.rb [hunk 1]
```diff
@@ -83,6 +83,15 @@ def next_page_redirect_path(page, log, current_user)
     end
   end
 
+  def next_page_interruption_screen?(page, log, current_user)
+    nxt_page_id = next_page(page, log, current_user)
+    nxt_page = get_page(nxt_page_id)
+
+    return unless nxt_page
+
+    nxt_page.questions.map(&:type).uniq.first == "interruption_screen"
```
Comment (kosiakkatrina) [inline app/models/form.rb]: Highly unlikely that we will have more than 1 question on the page for this but we could do something like:
`nxt_page.questions.any? { |question| question.type == "interruption_screen" }`

File: app/controllers/form_controller.rb [hunk 2]
```diff
@@ -133,7 +133,7 @@ def successful_redirect_path
       page_index = page_ids.index(@page.id)
       next_page = @log.form.next_page(@page, @log, current_user)
       previous_page = @log.form.previous_page(page_ids, page_index, @log, current_user)
-      if next_page.to_s.include?("value_check") || next_page == previous_page
+      if @log.form.next_page_interruption_screen?(@page, @log, current_user) || next_page == previous_page
```
Comment (kosiakkatrina) [inline app/controllers/form_controller.rb]: I like that it's a form method!

Comment (ghost) [reply app/controllers/form_controller.rb]: it's not anymore! https://github.com/communitiesuk/submit-social-housing-lettings-and-sales-data/pull/1215/files#diff-511b7d327a80c46192e9e7e6930bf51ae77f229ecf4b7e2771885d49e0d650fcR36

File: app/models/form.rb (hunk 1 above)
Comment (ghost) [reply app/models/form.rb]: moved to Page model and done something similar to what you suggested https://github.com/communitiesuk/submit-social-housing-lettings-and-sales-data/pull/1215/files#diff-511b7d327a80c46192e9e7e6930bf51ae77f229ecf4b7e2771885d49e0d650fcR36

File: app/controllers/form_controller.rb [hunk 3]
```diff
@@ -129,20 +129,26 @@ def is_referrer_check_answers?
...
       page_index = page_ids.index(@page.id)
-      next_page = @log.form.next_page(@page, @log, current_user)
-      previous_page = @log.form.previous_page(page_ids, page_index, @log, current_user)
-      if next_page.to_s.include?("value_check") || next_page == previous_page
-        return send("#{@log.class.name.underscore}_#{next_page}_path", @log, { referrer: "check_answers" })
+      next_page_id = form.next_page(@page, @log, current_user)
+      next_page = form.get_page(next_page_id)
+      previous_page = form.previous_page(page_ids, page_index, @log, current_user)
+
+      if next_page&.interruption_screen? || next_page_id == previous_page
```
Comment (ghost) [inline app/controllers/form_controller.rb:138]: the only logic change in this file is this line, we aren't any longer checking the page id to verify whether the page is an interruption screen but rather we're retrieving that value from the page itself

File: spec/fixtures/forms/2021_2022.json [hunk 4]
```diff
@@ -342,12 +342,14 @@
...
-                    "value": "Prefers not to say"
-                  }]
+                  "inferred_check_answers_value": [
+                    {
+                      "condition": {
+                        "armedforces": 3
+                      },
+                      "value": "Prefers not to say"
+                    }
+                  ]
```
Comment (ghost) [inline spec/fixtures/forms/2021_2022.json]: just new lines fixed by the linter

File: spec/fixtures/forms/2021_2022.json [hunk 5]
```diff
@@ -526,12 +528,14 @@
...
-                    "value": "Not known"
-                  }]
+                  "inferred_check_answers_value": [
+                    {
+                      "condition": {
+                        "postcode_known": 0
+                      },
+                      "value": "Not known"
+                    }
+                  ]
```
Comment (ghost) [inline spec/fixtures/forms/2021_2022.json]: just new lines fixed by the linter

File: app/models/form/sales/questions/household_wheelchair_check.rb [hunk 6]
```diff
@@ -9,15 +9,7 @@ def initialize(id, hsh, page)
...
-      "depends_on" => [
-        {
-          "wheel_value_check" => 0,
-        },
-        {
-          "wheel_value_check" => 1,
-        },
-      ],
-    }
+    @hidden_in_check_answers = true
```
Comment (kosiakkatrina) [inline app/models/form/sales/questions/household_wheelchair_check.rb]: Wound't we want to show it in check answers if it's not answered?

File: app/controllers/form_controller.rb (hunk 2 above)
Comment (kosiakkatrina) [reply app/controllers/form_controller.rb]: even better! 😂

File: app/models/form/sales/questions/household_wheelchair_check.rb (hunk 6 above)
Comment (natdeanlewissoftwire) [reply app/models/form/sales/questions/household_wheelchair_check.rb:12]: oh - I would have assumed what @kosiakkatrina did too - might have to change some other interruption screens I've made if this is the case

**Review by natdeanlewissoftwire — APPROVED** [review]
lgtm if we're sure about not showing these in the CYA page when unanswered

File: app/models/form/sales/questions/household_wheelchair_check.rb (hunk 6 above)
Comment (ghost) [reply app/models/form/sales/questions/household_wheelchair_check.rb]: updated!

**Pull request merged**